pkg/cosign/verify.go

```diff
@@ -133,8 +133,13 @@ type CheckOpts struct {
 	// Force offline verification of the signature
 	Offline bool
 
-	// TSACerts are the intermediate CA certs used to verify a time-stamping data.
-	TSACerts *x509.CertPool
+	// Set of flags to verify an RFC3161 timestamp used for trusted timestamping
+	// TSACertificate is the leaf TSA certificate. Optional, if provided in the timestamp
+	TSACertificate *x509.Certificate
```

Comment:
Shouldn't this be an array too ?

Reply:
Suggestion: I'd rename it to TSALeaveCertificates, or avoid using the term leaves to refer to the tsa certificate. I can be confusing.

Reply:
Yea, I'll make that update. Like with rekor pub keys, we just need to find the matching TSA certificate to verify against

Reply:
Thinking a bit more, I think it might be cleaner to just accept a single one for now.

There's a couple of issues:
* The expectation is that the TSA certificate is embedded in the timestamp, so it's likely you won't need to specify the TSA certificate
* If you specify a list of TSA certificates, what if none of them are the same as the cert embedded in the timestamp? Does this mean you don't trust what's in the timestamp? I think the only reason you should provide a TSA certificate is if it's missing from the timestamp, and that the root of trust for a TSA is the root certificates.
* There's no easy way to know which is the correct certificate to verify the timestamp without first parsing it, which I'd rather just let the verification library handle parsing after a successful verification. Otherwise, we need to run through all of the possible TSA certificates to try to find a successful verification

I think if we need to change the configuration in the future, it shouldn't be too hard, but I think we should just stick with one cert for now. Does that sound good?